docs/book/src/topics/managedcluster.md

```diff
@@ -398,3 +383,130 @@ spec:
+      maxUnavailable: 1
+    type: RollingUpdate
+  template:
+    osDisk:
+      diskSizeGB: 30
+      managedDisk:
+        storageAccountType: Premium_LRS
+      osType: Linux
+    sshPublicKey: ${AZURE_SSH_PUBLIC_KEY_B64:=""}
+    vmSize: ${AZURE_NODE_MACHINE_TYPE}
+---
+apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+kind: KubeadmConfig
+metadata:
+  name: ${CLUSTER_NAME}-vmss
+  namespace: default
+spec:
+  files:
+  - contentFrom:
+      secret:
+        key: worker-node-azure.json
+        name: ${CLUSTER_NAME}-vmss-azure-json
+    owner: root:root
+    path: /etc/kubernetes/azure.json
+    permissions: "0644"
+  - contentFrom:
+      secret:
+        key: value
+        name: ${CLUSTER_NAME}-kubeconfig
+    owner: root:root
+    path: /etc/kubernetes/admin.conf
+    permissions: "0644"  
+  joinConfiguration:
+    discovery:
+      file:
+        kubeConfigPath: /etc/kubernetes/admin.conf
+    nodeRegistration:
+      kubeletExtraArgs:
+        azure-container-registry-config: /etc/kubernetes/azure.json
+        cloud-provider: external
+      name: '{{ ds.meta_data["local_hostname"] }}'
+  preKubeadmCommands:
+  - kubeadm init phase upload-config all
+  ```
+
+### Installing Addons
+
+In order for the nodes to become ready, you'll need to install Cloud Provider Azure and a CNI.
+
+AKS will install Cloud Provider Azure on the self-managed nodes as long as they have the appropriate labels. You can add the required label on the nodes by running the following command on the AKS cluster:
+  
+```bash
+kubectl label node <node name> kubernetes.azure.com/cluster=<nodeResourceGroupName>
+```
+
+Repeat this for each node in the MachinePool.
+
+<aside class="note">
+
+<h1> Warning </h1>
+
+Note: CAPI does not currently support propagating labels from the MachinePool to the nodes, in the future this could be part of the MachinePool definition.
+
+</aside>
+
+For the CNI, you can install the CNI of your choice. For example, to install Azure CNI, run the following command on the AKS cluster:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-azure/main/templates/addons/azure-cni-v1.yaml
+```
+
+### Notes
+
+Some notes about how this works under the hood:
+
+- CAPZ will fetch the kubeconfig for the AKS cluster and store it in a secret named `${CLUSTER_NAME}-kubeconfig` in the management cluster.
+- You can customize the `MachinePool`, `AzureMachinePool`, and `KubeadmConfig` resources to your liking. The example above is just a starting point.
+- The `KubeadmConfig` resource will be used to generate a `kubeadm join` command that will be executed on each node in the VMSS. It uses the cluster kubeconfig for discovery. The `kubeadm init phase upload-config all` is run as a preKubeadmCommand to ensure that the kubeadm and kubelet configurations are uploaded to a ConfigMap. This step would normally be done by the `kubeadm init` command, but since we're not running `kubeadm init` we need to do it manually.
```

Comment:
Other than `preKubeadmCommands`, are there any other non-standard aspects of this config that enable joining to AKS that users should know should \*not\* be changed from the example?

Or more generally, I wonder if it might be helpful for this doc to describe how to take a "default" machine pool config and transform it into something that works in AKS in addition to (or instead of) the reverse here ("here's a machine pool that works for AKS, customize it however you need").

Reply:
Good question, everything that's in there is required (and defers from the otherwise example MachinePools we provide in other templates)

This part especially is important:

```
  joinConfiguration:
    discovery:
      file:
        kubeConfigPath: /etc/kubernetes/admin.conf
    nodeRegistration:
      kubeletExtraArgs:
        azure-container-registry-config: /etc/kubernetes/azure.json
        cloud-provider: external
      name: '{{ ds.meta_data["local_hostname"] }}'
```

along with:

```
files
  - contentFrom:
      secret:
        key: value
        name: ${CLUSTER_NAME}-kubeconfig
    owner: root:root
    path: /etc/kubernetes/admin.conf
    permissions: "0644"  
```

Reply:
Added a sentence to clarify